src/lib/Components/Text/__stories__/Typography.story.tsx

```diff
@@ -15,29 +15,30 @@ storiesOf("App Style/Typography")
     return <Serif>This is a blank serif</Serif>
   })
   .add("Typefaces", () => {
+    const fonts = {
+      "AGaramondPro Bold": GaramondBold,
+      "AGaramondPro BoldItalic": GaramondBoldItalic,
+      "AGaramondPro Italic": GaramondItalic,
+      "AGaramondPro Regular": GaramondRegular,
+      "AGaramondPro Semibold": GaramondSemibold,
+      "Avant Garde Gothic ITC": AvantGardeRegular,
+      "Unica77LL Italic": Unita77LLItalic,
+      "Unica77LL Medium": Unita77LLMedium,
+      "Unica77LL MediumItalic": Unita77LLMediumItalic,
+      "Unica77LL Regular": Unita77LLRegular,
+    }
     return (
       <Container>
         <Separator />
-        <GaramondBold>AGaramondPro Bold</GaramondBold>
-        <Separator />
-        <GaramondBoldItalic>AGaramondPro BoldItalic</GaramondBoldItalic>
-        <Separator />
-        <GaramondItalic>AGaramondPro Italic</GaramondItalic>
-        <Separator />
-        <GaramondRegular>AGaramondPro Regular</GaramondRegular>
-        <Separator />
-        <GaramondSemibold>AGaramondPro Semibold</GaramondSemibold>
-        <Separator />
-        <AvantGardeRegular>Avant Garde Gothic ITC</AvantGardeRegular>
-        <Separator />
-        <Unita77LLItalic>Unica77LL Italic</Unita77LLItalic>
-        <Separator />
-        <Unita77LLMedium>Unica77LL Medium</Unita77LLMedium>
-        <Separator />
-        <Unita77LLMediumItalic>Unica77LL MediumItalic</Unita77LLMediumItalic>
-        <Separator />
-        <Unita77LLRegular>Unica77LL Regular</Unita77LLRegular>
-        <Separator />
+        {Object.entries(fonts).map(value => {
+          const [name, Component] = value
+          return (
+            <React.Fragment key={name}>
+              <Component>{name}</Component>
+              <Separator />
+            </React.Fragment>
+          )
+        })}
```

Comment:
Credit for inspiring this idea (a dubious honour) goes to @jonallured. I still don't know if I _like_ this change, but I'm open to feedback 👍

Reply:
Woo! Converting things to data can be such a satisfying refactor! I think this is a little awkward because the extracted `fonts` local is an object rather than an array - what about something more like this:

```ts
const fonts = [
  { name: "AGaramondPro Bold", component: GaramondBold },
  { name: "AGaramondPro BoldItalic", component: GaramondBoldItalic },
  { name: "AGaramondPro Italic", component: GaramondItalic },
  { name: "AGaramondPro Regular", component: GaramondRegular },
  { name: "AGaramondPro Semibold", component: GaramondSemibold },
  { name: "Avant Garde Gothic ITC", component: AvantGardeRegular },
  { name: "Unica77LL Italic", component: Unita77LLItalic },
  { name: "Unica77LL Medium", component: Unita77LLMedium },
  { name: "Unica77LL MediumItalic", component: Unita77LLMediumItalic },
  { name: "Unica77LL Regular", component: Unita77LLRegular },
]
```

And then I'd use a few more locals to convert these to components:

```ts
const fontComponents = fonts.map((font, index) => {
  const Component = font.component
  return <Component key={index}>{font.name}</Component>
})

const separatedComponents = [<Separator />].concat(fontComponents.join(<Separator />))

return <Container>{separatedComponents}</Container>
```

Disclaimer: I didn't actually typecheck or lint this code so it's certainly possible I have syntax errors, etc - mostly just trying to convey an idea!! ❤️

Reply:
Oh yeah, that's way nicer. I also managed to get rid of a lot of boilerplate in this commit: https://github.com/artsy/emission/pull/1687/commits/0e93e52e7fa48952b75364f27eb732d3e5b4d844

Reply:
Here's what I've ended up with (JavaScript's `.join()` only takes strings, and outputs a string, sadly):

```tsx
const fonts = [
  { name: "AGaramondPro Bold", family: Fonts.GaramondBold },
  { name: "AGaramondPro BoldItalic", family: Fonts.GaramondBoldItalic },
  { name: "AGaramondPro Italic", family: Fonts.GaramondItalic },
  { name: "AGaramondPro Regular", family: Fonts.GaramondRegular },
  { name: "AGaramondPro Semibold", family: Fonts.GaramondSemibold },
  { name: "Avant Garde Gothic ITC", family: Fonts.AvantGardeRegular },
  { name: "Unica77LL Italic", family: Fonts.Unica77LLItalic },
  { name: "Unica77LL Medium", family: Fonts.Unica77LLMedium },
  { name: "Unica77LL MediumItalic", family: Fonts.Unica77LLMediumItalic },
  { name: "Unica77LL Regular", family: Fonts.Unica77LLRegular },
]
const fontComponents = fonts.map(({ name, family }) => {
  const Component = styled.Text`
    font-family: "${family}";
    font-size: 30px;
    margin-bottom: 10px;
  `
  return <Component key={name}>{name}</Component>
})
const separatedComponents = fontComponents.reduce(
  (memo, component, index) => [memo, <Separator key={index} />, component] as any // TypeScript doesn't know that JSX Elements can be arrays.
)
return (
  <Container>
    <Separator />
    {separatedComponents}
  </Container>
)
```

We end up with a bunch of nested arrays (based on `[memo, ...]`) but that should be fine for storybooks and for a limited number of fonts.

Reply:
LOL typescript! Come on computer you know what we mean!!